techSpec.js

```diff
@@ -0,0 +1,209 @@
+        db.each - executes code for each row in a dataset
+        db.get - executes code for the FIRST row - use when we know only one row will return. 
+
+d. Goals or Product and Technical Requirements
+
+    Product requirements: 
+        1. We should have a pre-populated SQLite database on the file system. 
+        2. Upon start-up of the Cat API Express server, it should establish a connection to the SQL database - or should it?  
+           Let's review:
+           The connection is required wherever we make a query via sqlite3 methods: all() get() and each()
+           The connection should be closed when we finish making the query and processing the result set. 
+           Therefore the connection does NOT need to be connected immediately upon server start-up.
+        3. The Cat API should be able to read from the SQLite database for GET all & GET by id requests and return objects. 
+        4. The Cat API should be able to write to the SQLite database for POST & PUT requests and return new/updated objects. 
+        4. The Cat API should be able to delete from the SQLite database for DEL requests. 
+        
+
+    Technical requirements: 
+        
+        1.
+        Create and populate 'cat' & 'pedigree breed' tables via sqlite in Terminal. 'CREATE TABLE' and so forth....
+        Remove list of cats and breed objects from catsRepo.js & breedsRepo.js respectively.   
+        Review if the repositories still need to be classes in this case? 
+        Note that id to have PRIMARY KEY constraint (needs to be unique)  
+         
+        2. 
+        Install sqlite3 node module into the root directory of cat api.
+        Import/require into repository files for use - Eg catsRepo.js is where all direct interaction with database happens. 
+        Repository methods will call a third party function that starts a database connection, queries the database and closes connection. 
+        These function should take three arguments: database path ('./catDb' or './breedDb') request type (GET / PUT / POST / DEL ), SQL query as a string. 
+        This function can be an seperated imported file/module for separation of concerns, and reusability by Breed repository 
+        
+        +++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+        Each HTTP request (GET, POST, PUT, DEL) requires a different database query command. 
+        How should we deal with this?
+        -----------------------------
+
+        Either we have one central 'database connector' function with multiple cases, HTTP request type is passed in as argument, Eg:
+
+        const databaseConnection = ( <database path>, <request type>,  <sql command>) => { 
+            if (<request type> === GET) {...} etc  
+        } 
+
+        OR each request has its own 'database connector' function. This is simplest as request types pass different parameters as arguments.
```

Comment:
I think this approach would be preferable. Each function in `CatRepository` could be altered to run its own query on the open database connection. This has the advantage that it keeps each function small, as with one central function that would grow linearly with the number of queries and could become hard to maintain if we added in 20+ routes.